### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by zateshilov-k**
beta of hospital epam web application.

### CODE REVIEW

File: src/main/java/com/epam/lab/hospitalspring/controller/PatientController.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            model.addAttribute("patient", currentPatient);
+            Gson gson = GsonFactory.buildGson(locale);
+            model.addAttribute("diagnoses", gson.toJson(currentPatient.getDiagnosisList()));
+            return "patientDiagnosisCard";
+        } else {
+            return "redirect:/error/errorMessage";
+        }
+    }
+
+    @GetMapping(value = "/addPatient")
```
Comment (Ilya21) [inline src/main/java/com/epam/lab/hospitalspring/controller/PatientController.java:84]: what is this method for?

File: src/main/java/com/epam/lab/hospitalspring/controller/PatientController.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    public String getAddPatientPage(Patient patient) {
+        return "patient";
+    }
+
+    @PostMapping(value = "/addPatient")
```
Comment (Ilya21) [inline src/main/java/com/epam/lab/hospitalspring/controller/PatientController.java:89]: Don't use verb in uri if you follow rest api. It should be something like written below
localhost / patients - post if you create, put for update, get for getting, delete for deleting

File: src/main/java/com/epam/lab/hospitalspring/controller/PersonalController.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        if (personalService.update(personalForm, id)) {
+            return "redirect:/personals";
+        } else {
+            return "redirect:/personal/" + id + "?loginAleadyUsed";
+        }
+    }
+
+    @GetMapping("/deletePersonalFromDB/{id}")
+    public String deletePersonalFromDB(@PathVariable("id") Long id) {
+        personalService.deleteFromDB(id);
```
Comment (Ilya21) [inline src/main/java/com/epam/lab/hospitalspring/controller/PersonalController.java:62]: Just `delete`, FromDb is useless i think
can you delete from somewhere else?)

File: src/main/java/com/epam/lab/hospitalspring/security/details/PersonalDetailsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+@Service
+public class PersonalDetailsServiceImpl implements UserDetailsService {
+    @Autowired
+    PersonalRepository personalRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String login) throws UsernameNotFoundException {
+        Optional<Personal> personal = personalRepository.findOneByLogin(login);
+        if (personal.isPresent()) {
```
Comment (Ilya21) [inline src/main/java/com/epam/lab/hospitalspring/security/details/PersonalDetailsServiceImpl.java:21]: Using isPresent  is not good practice.
You should use return personal.orElseThrow() for example. It's made code cleaner

File: src/main/java/com/epam/lab/hospitalspring/service/impl/PatientServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        boolean isValidData = true;
+        if (patientForm.getFirstName().equals("") || patientForm.getFirstName() == null
+                || patientForm.getLastName().equals("") || patientForm.getLastName() == null) {
+            isValidData = false;
+        }
+        return isValidData;
+    }
+
+    @Override
+    public void addPatient(Patient patient) {
```
Comment (Ilya21) [inline src/main/java/com/epam/lab/hospitalspring/service/impl/PatientServiceImpl.java:33]: oh, and here you are using GoF pattern builder, don't forget say about this :)

File: src/main/java/com/epam/lab/hospitalspring/util/GsonFactory.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package com.epam.lab.hospitalspring.util;
+
+import com.epam.lab.hospitalspring.model.Diagnosis;
+import com.epam.lab.hospitalspring.model.Prescription;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.util.Locale;
+
+public class GsonFactory {
```
Comment (Ilya21) [inline src/main/java/com/epam/lab/hospitalspring/util/GsonFactory.java:10]: And factory pattern)

File: src/main/resources/templates/patientDiagnosisCard.html [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+        currentDiagnosisRow = $(this).index();
+        var currentDiagnosisId = this.cells[0].innerHTML;
+        updateDiagnosisElementsStyle();
+        var doPrescriptionButton = document.getElementById("doPrescription");
+        doPrescriptionButton.style.display = "none";
+        var prescriptionSubmitButton = document.getElementById("prescriptionSubmit");
+        prescriptionSubmitButton.value = currentDiagnosisId;
+        updatePrescriptionsTable(currentDiagnosisId);
+    }
+    function updatePrescriptionsTable(diagnosis_id) {
```
Comment (Ilya21) [inline src/main/resources/templates/patientDiagnosisCard.html:248]: Don't forget say about using js

**Comment by Ilya21** [comment]
Nice job guys!

File: src/main/java/com/epam/lab/hospitalspring/controller/PatientController.java (hunk 1 above)
Comment (zateshilov-k) [reply src/main/java/com/epam/lab/hospitalspring/controller/PatientController.java:84]: @Ilya21 
Метод для отображения ошибок через аннотации в @Entity.
https://spring.io/guides/gs/validating-form-input/
"It is marked as being backed up by the personForm object that you saw in the GET method in the web controller. This is known as a bean-backed form."

**Pull request merged**